Nightly rust nightly builds are broken

Despite #24557, the nightly build that uses the nightly rust compiler is broken for reasons I don't understand: https://tools.taskcluster.net/tasks/aOfB7Z_fS8eNUeOzNUdfSw

```
[0m[1m[38;5;9merror[E0463][0m[0m[1m: can't find crate for `rustc`[0m
[0m  [0m[0m[1m[38;5;12m--> [0m[0mcomponents/script_plugins/lib.rs:23:1[0m
[0m   [0m[0m[1m[38;5;12m|[0m
[0m[1m[38;5;12m23[0m[0m [0m[0m[1m[38;5;12m| [0m[0mextern crate rustc;[0m
[0m   [0m[0m[1m[38;5;12m| [0m[0m[1m[38;5;9m^^^^^^^^^^^^^^^^^^^[0m[0m [0m[0m[1m[38;5;9mcan't find crate[0m

[0m[1m[38;5;9merror[0m[0m[1m: aborting due to previous error[0m
```

--- a/etc/taskcluster/decision_task.py
+++ b/etc/taskcluster/decision_task.py
@@ -267,7 +267,7 @@ def with_rust_nightly():
         modified_build_env["RUSTFLAGS"] = " ".join(flags)
 
     return (
-        linux_build_task("with Rust Nightly", build_env=modified_build_env)
+        linux_build_task("with Rust Nightly", build_env=modified_build_env, install_rustc_dev=False)
         .with_treeherder("Linux x64", "RustNightly")
         .with_script("""
             echo "nightly" > rust-toolchain
@@ -748,8 +748,8 @@ def macos_task(name):
     )
 
 
-def linux_build_task(name, *, build_env=build_env):
-    return (
+def linux_build_task(name, *, build_env=build_env, install_rustc_dev=True):
+    task = (
         linux_task(name)
         # https://docs.taskcluster.net/docs/reference/workers/docker-worker/docs/caches
         .with_scopes("docker-worker:cache:servo-*")
@@ -766,9 +766,11 @@ def linux_build_task(name, *, build_env=build_env):
         .with_env(**build_env, **unix_build_env, **linux_build_env)
         .with_repo()
         .with_script("rustup set profile minimal")
-        # required by components/script_plugins:
-        .with_script("rustup component add rustc-dev")
     )
+    if install_rustc_dev:
+        # required by components/script_plugins:
+        task = task.with_script("rustup component add rustc-dev")
+    return task
 
 
 def android_build_task(name):

--- a/etc/taskcluster/decision_task.py
+++ b/etc/taskcluster/decision_task.py
@@ -271,6 +271,7 @@ def with_rust_nightly():
         .with_treeherder("Linux x64", "RustNightly")
         .with_script("""
             echo "nightly" > rust-toolchain
+            rustup component add rustc-dev
             ./mach build --dev
             ./mach test-unit
         """)

